packages/market-service/src/coingecko/coingecko.ts

```diff
@@ -37,69 +36,68 @@ type CoinGeckoAssetData = {
+  private readonly maxCount = 2500
+  private readonly maxPerPage = 250
+
   baseUrl: string
   APIKey: string
 
-  constructor(args: { coinGeckoAPIKey: string }) {
-    const { coinGeckoAPIKey } = args
-    this.APIKey = coinGeckoAPIKey
+  constructor(args: CoinGeckoMarketServiceArgs) {
+    this.APIKey = args.coinGeckoAPIKey
     // if we have a key - use the pro- api
     this.baseUrl = `https://${this.APIKey ? 'pro-' : ''}api.coingecko.com/api/v3`
   }
 
-  private readonly defaultGetByMarketCapArgs: FindAllMarketArgs = {
-    count: 2500
-  }
-
   private maybeAddAPIKey = (): string => {
     return this.APIKey ? `&x_cg_pro_api_key=${this.APIKey}` : ''
   }
 
   async findAll(args?: FindAllMarketArgs) {
-    const argsToUse = { ...this.defaultGetByMarketCapArgs, ...args }
-    const { count } = argsToUse
-    const perPage = count > 250 ? 250 : count
+    const count = args?.count ?? this.maxCount
+    const perPage = count < this.maxPerPage ? count : this.maxPerPage
     const pages = Math.ceil(bnOrZero(count).div(perPage).toNumber())
 
     const urlAtPage = (page: number) =>
       `${
         this.baseUrl
       }/coins/markets?vs_currency=usd&order=market_cap_desc&per_page=${perPage}&page=${page}&sparkline=false${this.maybeAddAPIKey()}`
+
     const pageCount = Array(pages)
       .fill(0)
       .map((_v, i) => i + 1)
+
     try {
-      const combined = (
-        await Promise.all(
-          pageCount.map(async (page) => axios.get<CoinGeckoMarketCap>(urlAtPage(page)))
-        )
-      ).flat()
-      return combined
-        .map(({ data }) => data ?? []) // filter out rate limited results
-        .flat()
-        .sort((a, b) => (a.market_cap_rank > b.market_cap_rank ? 1 : -1))
-        .reduce((acc, cur) => {
-          const { id } = cur
-          try {
-            const assetId = adapters.coingeckoToAssetId(id)
-            if (!assetId) return acc
-            const curWithoutId = omit(cur, 'id') // don't leak this through to clients
-            acc[assetId] = {
-              price: curWithoutId.current_price.toString(),
-              marketCap: curWithoutId.market_cap.toString(),
-              volume: curWithoutId.total_volume.toString(),
-              changePercent24Hr: curWithoutId.price_change_percentage_24h,
-              supply: curWithoutId.circulating_supply.toString(),
-              maxSupply: curWithoutId.max_supply
-                ? curWithoutId.max_supply.toString()
-                : curWithoutId.total_supply?.toString()
-            }
-            return acc
-          } catch {
-            return acc // no AssetId found, we don't support this asset
-          }
-        }, {} as MarketCapResult)
+      const marketData = await Promise.all(
+        pageCount.map(async (page) => {
+          const { data } = await axios.get<CoinGeckoMarketCap>(urlAtPage(page))
+          return data ?? []
+        })
+      )
+
+      return marketData.flat().reduce<MarketCapResult>((prev, asset) => {
```

Comment:
we lost the explicit sort, but i guess if we're calling the `market_cap_desc` endpoint we can safely assume they're sorted.

Reply:
yeah that was the intention as it was already sorted and there was a ton of extra chaining before that was doing essentially nothing